File: src/main/java/com/test/Application/entity/Timetable.java
package com.test.Application.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Timetable implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @OneToOne
    private Room room;
    @OneToOne
    private Item item;
    @OneToOne
    private Teacher teacher;
    private Date date;
}

File: appForSchool-front/src/app/entity/timetable.ts
import {Room} from "./room";
import {Item} from "./item";
import {Teacher} from "./teacher";

export class Timetable{
  id!: number;
  room: Room;
  item: Item;
  teacher: Teacher;
  date: Date;

  constructor() {
    this.room = new Room();
    this.item = new Item();
    this.teacher = new Teacher();
    this.date = new Date();
  }
}
